examples/lang/erc721/src/lib.rs

```diff
@@ -0,0 +1,453 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use core::result::Result;
+use ink_core::{
+    env::DefaultSrmlTypes,
+    memory::format,
+    storage,
+};
+use ink_lang::contract;
+use ink_model;
+use scale::{
+    Decode,
+    Encode,
+};
+
+pub type EnvHandler = ink_model::EnvHandler<ink_core::env::ContractEnv<DefaultSrmlTypes>>;
+pub type TokenId = u32;
+pub type Counter = u32;
+
+#[cfg(feature = "ink-generate-abi")]
+use type_metadata::Metadata;
+
+#[derive(Encode, Decode, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "ink-generate-abi", derive(Metadata))]
+pub enum Error {
+    NotOwner,
+    NotApproved,
+    TokenExists,
+    TokenNotFound,
+    CanNotInsert,
+    CanNotRemove,
+    CanNotGetCounter,
+    AccountZeroNotAllowed,
+}
+
+impl From<Error> for u32 {
+    fn from(item: Error) -> Self {
+        item as u32
+    }
+}
```

Comment:
I would generally not implement such `From` impls for primitive types.
Maybe I am a bit too over-restrictive here: What I would do instead is to simply provide a method on `Error`: `fn to_u32(self) -> u32` and also I would derive `Copy` (and `Clone`) for `Error` because we can.